pkg/apis/provisioning/v1alpha5/provisioner_validation.go

```diff
@@ -79,36 +78,13 @@ func (c *Constraints) validateLabels() (errs *apis.FieldError) {
 		for _, err := range validation.IsValidLabelValue(value) {
 			errs = errs.Also(apis.ErrInvalidValue(fmt.Sprintf("%s, %s", value, err), fmt.Sprintf("labels[%s]", key)))
 		}
-		if RestrictedLabels.Has(key) {
-			errs = errs.Also(apis.ErrInvalidKeyName(key, "labels", "label is restricted"))
-		}
-		if _, ok := WellKnownLabels[key]; !ok && IsRestrictedLabelDomain(key) {
-			errs = errs.Also(apis.ErrInvalidKeyName(key, "labels", "label domain not allowed"))
+		if err := IsRestrictedLabel(key); err != nil {
+			errs = errs.Also(apis.ErrInvalidKeyName(key, "labels", fmt.Sprintf("%s", err)))
```

Comment:
Small nit, could just be:
`errs = errs.Also(apis.ErrInvalidKeyName(key, "labels", err.Error()))`

Reply:
Good catch. Done.